substrate/frame/staking-async/ah-client/src/lib.rs

```diff
@@ -338,19 +375,19 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type ValidatorSetAppliedAt<T: Config> = StorageValue<_, SessionIndex, OptionQuery>;
 
-	/// Stores offences that have been received while the pallet is in [`OperatingMode::Buffered`]
-	/// mode.
+	/// Offences collected while in [`OperatingMode::Buffered`] mode.
+	///
+	/// These are temporarily stored and sent once the pallet switches to [`OperatingMode::Active`].
+	/// For each offender, only the highest `slash_fraction` is kept.
 	///
-	/// These offences are collected and buffered for later processing when the pallet transitions
-	/// to [`OperatingMode::Active`]. This allows the system to defer slashing or reporting logic
-	/// until communication with the counterpart pallet on AssetHub is fully established.
+	/// Internally stores as a nested BTreeMap:
+	/// `session_index -> (offender -> (reporter, slash_fraction))`.
 	///
-	/// This storage is only used in `Buffered` mode; in `Active` mode, offences are immediately
-	/// sent, and in `Passive` mode, they are delegated to the [`Config::Fallback`] implementation.
+	/// Note: While the [`rc_client::Offence`] type includes a list of reporters, in practice there
+	/// is only one. In this pallet, we assume this is the case and store only the first reporter.
 	#[pallet::storage]
 	#[pallet::unbounded]
-	pub type BufferedOffences<T: Config> =
-		StorageValue<_, Vec<(SessionIndex, Vec<rc_client::Offence<T::AccountId>>)>, ValueQuery>;
+	pub type BufferedOffences<T: Config> = StorageValue<_, BufferedOffencesMap<T>, ValueQuery>;
```

Comment:
Just a note that we don't need to have migration for these as they are never meant to be set unless the pallet is in buffered mode (which it shouldn't be at the time of a runtime upgrade). 

Additionally, this pallet hasn’t been deployed to Kusama or Polkadot yet, so migration can be safely skipped.